components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/PasswordRecoveryApiV2.java

```diff
@@ -0,0 +1,196 @@
+import org.wso2.carbon.identity.mgt.endpoint.util.client.model.passwordrecovery.v2.RecoveryInitRequest;
+import org.wso2.carbon.identity.mgt.endpoint.util.client.model.passwordrecovery.v2.RecoveryRequest;
+import org.wso2.carbon.identity.mgt.endpoint.util.client.model.passwordrecovery.v2.RecoveryResponse;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * API related to password recovery with user preferred channel.
+ */
+public class PasswordRecoveryApiV2 {
+
+    String basePath = IdentityManagementEndpointUtil.buildEndpointUrl(IdentityManagementEndpointConstants
+            .UserInfoRecovery.RECOVERY_API_V2_RELATIVE_PATH);
+    private ApiClient apiClient;
+
+    public PasswordRecoveryApiV2() {
+
+        this(Configuration.getDefaultApiClient());
+    }
+
+    public PasswordRecoveryApiV2(ApiClient apiClient) {
+
+        this.apiClient = apiClient;
+    }
+
+    public ApiClient getApiClient() {
+
+        return apiClient;
+    }
+
+    public void setApiClient(ApiClient apiClient) {
+
+        this.apiClient = apiClient;
+    }
+
+    /**
+     * This API can be used to initiate recovering forgotten password.
+     *
+     * @param recoveryInitRequest Password recovery initiating request. (required)
+     * @param tenantDomain        Tenant Domain which user belongs. Default &#x60;carbon.super&#x60; (optional)
+     * @param headers             If reCaptcha respond is found, embedded in request header. (optional)
+     * @return Account recovery options response object.
+     * @throws ApiException If fails to make API call.
+     */
+    public List<AccountRecoveryType> initiatePasswordRecovery(RecoveryInitRequest recoveryInitRequest,
+                                                              String tenantDomain, Map<String, String> headers)
+            throws ApiException {
+
+        String localVarPath = "/password/init".replaceAll("\\{format\\}", "json");
+        return initiateRecovery(recoveryInitRequest, tenantDomain, null, localVarPath);
+    }
+
+    /**
+     * This API is used to recover password via selected recovery option.
+     *
+     * @param recoveryRequest   Recovery request. (required)
+     * @param tenantDomain      Tenant Domain which user belongs. Default &#x60;carbon.super&#x60; (optional)
+     * @param headers           Any additional headers to be embedded. (optional)
+     * @return Recovery response.
+     * @throws ApiException If fails to make API call.
+     */
+    public RecoveryResponse recoverPassword(RecoveryRequest recoveryRequest, String tenantDomain,
+                                            Map<String, String> headers) throws ApiException {
+
+        String localVarPath = "/password/recover".replaceAll("\\{format\\}", "json");
+        return recover(recoveryRequest, tenantDomain, headers, localVarPath);
+    }
+
+    /**
+     * This API can be used to initiate recovering forgotten password/username.
```

Comment:
`initiateRecovery` is a private method and it resides on the `PasswordRecoveryApiV2` class. Is there any way to use this for username recovery?

Reply:
Please check the `private RecoveryResponse recover(` method as well.

Reply:
It can be used for username recovery. Hence, renamed the class to RecoveryApiV2, so that we can use it for both password and username recovery.